### PR TITLE
Fixed issue 375. Code was already working, just needed some bug fixes…

### LINKED ISSUE #375: Enable support for a single entity connection string
I have a test Service Bus deployed with only one queue. Connecting to only the queue entity using Service Bus Explorer 4.1.113 works without any issues. Connecting to the queue and say the topic entities, generates the error shown below. This issue occurs with any entity combination selected when connecting to a queue, topic, et cetera directly.

```
<09:41:29> The application is now connected to the sb://some-service-bus-name.servicebus.windows.net/ service bus namespace.
<09:41:29> MessagingFactory successfully created
<09:41:29> The queue some-queue-name has been successfully retrieved.
<09:41:29> Failed to retrieve Service Bus topics. Exception: Microsoft.ServiceBus.Messaging.MessagingException: Cannot get entity 'some-queue-name' because it is not of type TopicDescription. Check that you are using method(s) with the correct entity type. ---> System.Runtime.Serialization.SerializationException: Error in line 1 position 2641. Expecting element 'TopicDescription' from namespace 'http://schemas.microsoft.com/netservices/2010/10/servicebus/connect'.. Encountered 'None'  with name '', namespace ''. 

              at System.Runtime.Serialization.DataContractSerializer.InternalReadObject(XmlReaderDelegator xmlReader, Boolean verifyObjectName, DataContractResolver dataContractResolver)

              at System.Runtime.Serialization.XmlObjectSerializer.ReadObjectHandleExceptions(XmlReaderDelegator reader, Boolean verifyObjectName, DataContractResolver dataContractResolver)

              at Microsoft.ServiceBus.Messaging.MessagingDescriptionSerializer`1.DeserializeFromAtomFeed(Stream stream)

              at Microsoft.ServiceBus.Messaging.ServiceBusResourceOperations.GetAsyncResult`1.<GetAsyncSteps>d__22.MoveNext()

              --- End of inner exception stack trace ---

              at Microsoft.Azure.ServiceBusExplorer.ServiceBusHelper.GetTopicUsingEntityPath() in C:\projects\servicebusexplorer\src\ServiceBusExplorer\Helpers\ServiceBusHelper.cs:line 2087

              at Microsoft.Azure.ServiceBusExplorer.ServiceBusHelper.GetTopics(String filter) in C:\projects\servicebusexplorer\src\ServiceBusExplorer\Helpers\ServiceBusHelper.cs:line 2062

              at Microsoft.Azure.ServiceBusExplorer.Forms.MainForm.GetEntities(EntityType entityType) in C:\projects\servicebusexplorer\src\ServiceBusExplorer\Forms\MainForm.cs:line 4477
```

The expected behavior in this specific example is that the connection is made without an error, the queue icon should be shown with the queue listed underneath and the topic icon should also be shown, but without any topics listed underneath.

### PR DESCRIPTION
**Pull request by rajrao**
Fixes #375 
This looks like its already working correctly. I had to remove the entitySpecific check and also fix an issue with FilterOutException (which must have happened when the code was changed to be async). After that it works! The only thing is that it spits out an ugly error as it spits out the whole stack (not just the message). Thought of changing that, but the entire stack can be nice!

Tested both topic and queue based connection strings
Endpoint=sb://xxxx.servicebus.windows.net/;SharedAccessKeyName=test;SharedAccessKey=abcd;EntityPath=topicPath

### CODE REVIEW

**Comment by SeanFeldman** [comment]
:white_check_mark: [Build ServiceBusExplorer 4.2.0-Issue375-EnableSupportSingleEntityConnStr.54.build.638 completed](https://ci.appveyor.com/project/seanfeldman/servicebusexplorer/builds/30859441) (commit https://github.com/paolosalvatori/ServiceBusExplorer/commit/8eff24e4e1 by @rajrao)

File: src/ServiceBusExplorer/Forms/ConnectForm.cs [hunk 1]
```diff
@@ -244,17 +244,6 @@ private void btnOk_Click(object sender, EventArgs e)
                     MainForm.StaticWriteToLog(ConnectionStringCannotBeNull);
                     return;
                 }
-                if (ConnectionStringHelper.IsEntitySpecific(txtUri.Text))
```
Comment (SeanFeldman) [inline src/ServiceBusExplorer/Forms/ConnectForm.cs]: In case `ConnectionStringHelper.IsEntitySpecific()` is no longer used then it could be removed.

Comment (rajrao) [reply src/ServiceBusExplorer/Forms/ConnectForm.cs]: Done!

**Comment by SeanFeldman** [comment]
:white_check_mark: [Build ServiceBusExplorer 4.2.0-Issue375-EnableSupportSingleEntityConnStr.55.build.639 completed](https://ci.appveyor.com/project/seanfeldman/servicebusexplorer/builds/30866939) (commit https://github.com/paolosalvatori/ServiceBusExplorer/commit/e8b6960551 by @rajrao)

**Pull request merged**